repository.go

```diff
@@ -453,3 +454,129 @@ func (r *Repository) Reference(name plumbing.ReferenceName, resolved bool) (
 func (r *Repository) References() (storer.ReferenceIter, error) {
 	return r.s.IterReferences()
 }
+
+// ResolveRevision resolves revision to corresponding hash
+func (r *Repository) ResolveRevision(rev plumbing.Revision) (*plumbing.Hash, error) {
+	p := revision.NewParserFromString(string(rev))
```

Comment:
this does not make a lot of sense.
You have a revision, turn it into a string, then call NewParserFromString.
Why don't you create a parser from the revision directly: `p := rev.NewParser()`.